packages/react/src/WalletContext.tsx

```diff
@@ -57,11 +57,21 @@ export const WalletContextProvider: React.FC<React.PropsWithChildren> = (
     });
   }, [setup]);
 
+  const onCloseModal = (): void => {
+    setIsWaitingForConnection(false);
+  };
+
+  if (typeof window !== 'undefined') {
+    const closeButton = document?.getElementsByClassName('close-button')[0];
```

Comment:
Not a bad idea to use the DOM directly like this, but would have to ensure the node is there before binding the event listener

Reply:
Wait, does this actually work?  I wonder if getting a re-render from state update 🤔  cool.

Reply:
Added a warning just in case :)